prometheus-openstack-exporter

```diff
@@ -616,29 +617,60 @@ class Nova:
             elif not isinstance(cpu_info, dict):
                 cpu_info = json.loads(cpu_info)
                 arch = cpu_info["arch"]
-            label_values = [
-                config["cloud"],
-                host,
-                self.aggregate_map.get(host, "unknown"),
-                self.services_map[host],
-                arch,
-            ]
-            # Disabled hypervisors return None below, convert to 0
-            vms.labels(*label_values).set(squashnone(h["running_vms"]))
-            vcpus_total.labels(*label_values).set(squashnone(h["vcpus"]))
-            vcpus_used.labels(*label_values).set(squashnone(h["vcpus_used"]))
-            mem_total.labels(*label_values).set(squashnone(h["memory_mb"]))
-            mem_used.labels(*label_values).set(squashnone(h["memory_mb_used"]))
-            disk_total.labels(*label_values).set(squashnone(h["local_gb"]))
-            disk_used.labels(*label_values).set(squashnone(h["local_gb_used"]))
-
-            if config.get("schedulable_instance_size", False):
-                schedulable_instances.labels(*label_values).set(
-                    self._get_schedulable_instances(h)
-                )
-                schedulable_instances_capacity.labels(*label_values).set(
-                    self._get_schedulable_instances_capacity(h)
-                )
+
+            for agg in self.prodstack["aggregates"]:
+                agg_key = host + "_" + agg["name"]
+                if self.aggregate_map.get(agg_key) is None:
+                    continue
+                label_values = [
+                    config["cloud"],
+                    host,
+                    self.aggregate_map[agg_key],
+                    self.services_map[host],
+                    arch,
+                ]
+                # Disabled hypervisors return None below, convert to 0
+                vms.labels(*label_values).set(squashnone(h["running_vms"]))
+                vcpus_total.labels(*label_values).set(squashnone(h["vcpus"]))
+                vcpus_used.labels(*label_values).set(squashnone(h["vcpus_used"]))
+                mem_total.labels(*label_values).set(squashnone(h["memory_mb"]))
+                mem_used.labels(*label_values).set(squashnone(h["memory_mb_used"]))
+                disk_total.labels(*label_values).set(squashnone(h["local_gb"]))
+                disk_used.labels(*label_values).set(squashnone(h["local_gb_used"]))
+
+                if config.get("schedulable_instance_size", False):
+                    schedulable_instances.labels(*label_values).set(
+                        self._get_schedulable_instances(h)
+                    )
+                    schedulable_instances_capacity.labels(*label_values).set(
+                        self._get_schedulable_instances_capacity(h)
+                    )
+
+            # No aggregates found, log the metric for the host
+            if not self.prodstack:
```

Comment:
I might be misunderstanding it, but I think here we want to be checking if this host is not in an aggregate?  Currently this will only eval to true if there are no aggregates, but the existing logic will add it if the host is not in an aggregate.  

Maybe the logic could be combined with the previous loop, like:

```
at_least_one_aggregate = False
for agg in self.prodstack["aggregates"]:
  ...
  if self.aggregate_map.get(agg_key) is None:
    continue
  at_least_one_aggregate = True
  ...

if not at_least_one_aggregate:
  label_values = [
  ...
```

Reply:
Ah, I can see the issue. I thought if the aggregate_map is empty then I can assume the host isn't in any aggregate. But a different host could be in aggregate (making aggregate_map non-empty).

Reply:
Modified as suggested. Please check again.